app/src/main/java/com/example/playlistmaker/SearchHistory.kt

```diff
@@ -0,0 +1,54 @@
+package com.example.playlistmaker
+
+import android.content.SharedPreferences
+import com.example.playlistmaker.dataclasses.ITunesTrack
+import com.google.gson.Gson
+import com.google.gson.reflect.TypeToken
+
+class SearchHistory(val sharedPreferences: SharedPreferences) {
+
+    private var trackList: MutableList<ITunesTrack> = mutableListOf()
+    fun read(): MutableList<ITunesTrack> {
```

Comment:
```suggestion
    fun read(): List<ITunesTrack> {
```
Лучше всегда передавать копию списка под интерфейсом `List`, чтобы снаружи нельзя было этот список поменять, т.к. это может сломать логику в этом классе